### PR TITLE
Plotting

### PR DESCRIPTION
**Pull request by madzhun**
Could you please take a look and comment the format, whether you want to do plotting this way, and do you need it during the execution

### CODE REVIEW

**Comment by AlexandreBeaulne** [comment]
great start

can you please rename to `plot.py`

File: plotting.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import os
```
Comment (AlexandreBeaulne) [inline plotting.py]: trailing whitespace

File: plotting.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import matplotlib.pyplot as plt
+import os 
+import glob
+
+def plot_scores_all(save=False):
+    for filename in glob.glob('*.csv'):
+        print(filename)
+        base = os.path.basename(filename)
+        game_csv = os.path.splitext(base)[0]    # extracting filename without extension
+
```
Comment (AlexandreBeaulne) [inline plotting.py]: trailing whitespace

File: plotting.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+def plot_scores_all(save=False):
+    for filename in glob.glob('*.csv'):
+        print(filename)
+        base = os.path.basename(filename)
+        game_csv = os.path.splitext(base)[0]    # extracting filename without extension
+    
+        ep_scores = np.loadtxt(filename, delimiter=',', skiprows=1)
+        plt.figure()
+        plt.plot(ep_scores[:, 0], ep_scores[:, 1], label='Pong-v0')
```
Comment (AlexandreBeaulne) [inline plotting.py]: extract the game name (`Pong-v0`) from the CSV filename instead of hardcoding it here

File: plotting.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        plt.xlabel('Episodes')
+        plt.ylabel('Score per episode')
+        plt.legend()
+        if save:
+            plt.savefig('{}.png'.format(game_csv))
+        plt.show()
+
+
+def plot_scores_graph(game, save=False):
```
Comment (AlexandreBeaulne) [inline plotting.py]: lots of overlap between this function and `plot_score_all()`

File: plotting.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    plt.plot(ep_scores[:, 0], ep_scores[:, 1], label=game)
+    plt.xlabel('Episodes')
+    plt.ylabel('Score per episode')
+    plt.legend()
+    if save:
+        plt.savefig('{}_episode_scores.png'.format(game))
+    plt.show()
+
+
+def plot_durations(scores):      # might help during execution, needs modification
```
Comment (AlexandreBeaulne) [inline plotting.py]: this function is broken (undefined variables, using libraries not imported, commented out code), fix it or remove it

File: plotting.py (hunk 1 above)
Comment (madzhun) [reply plotting.py]: don't know what to do with whitespacing

**Comment by madzhun** [comment]
- Removed plot_durations() for a moment, need to do it in code to make it work
- renamed
- fixed plot_scores_all()

**Comment by madzhun** [comment]
anything else?

**Comment by AlexandreBeaulne** [comment]
> anything else?

no approved please merge

**Pull request merged**